CONTRIBUTING.md

```diff
@@ -0,0 +1,190 @@
+
+1. [Find](https://github.com/archway-network/archway/issues) or [Open](https://github.com/archway-network/archway/issues/new) an issue;
+2. __Engage__ in _constructive_ and _thoughful_ __discussion__ on the issue;
+3. __Coordinate__ with _other contributors_ on the issue;
+4. __Follow__ _GitHub best practices_ and the _guidelines_ in the issue templates;
+
+This repository uses the __good first issue__ label to identify issues suitable for first-time contributors.
+
+Ensure a proper title for the issue that clearly defines what the issue is. This will greatly assist others to easily find related issues and avoid duplication.
+
+Make use of the following labels to provide additional context:
+
+- _bug_: Something isn't working;
+- _documentation_: Improvements or additions to documentation;
+- _enhancement_: New feature or request;
+- _question_: Further information is requested;
+
+Example: `Lack of Community Guidelines` <- label as `documentation`
+
+## 3. Branching
+
+The `main` branch represents the latest development features and fixes. Thus, it will most likely not be compatible with running networks such as `mainnet` or any testnets.
+
+Releases are found under `release/#.#.x` using a semantic versioning scheme {braking}.{feature}.{fix} denoting each minor version.
+
+The `main` branch and `release/*` branches are protected and can only be updated via a Pull Request. Thus, do not make any commits to these branches directly.
+
+For increased manageability please address one issue per branch and use the following branch name format:
+
+`<user>/<issue#>-<short_description>`
+
+## 4. Testing
+
+All _features_ and _fixes_ must be accompanied by appropriate unit/integration tests. Please ensure to follow Go and CosmosSDK best practices regarding tests. Archway utilizes table-driven tests, as explained in [this blogpost](https://dave.cheney.net/2019/05/07/prefer-table-driven-tests) by Dave Cheney.
+
+Simple rules for table-driven tests:
+
+1. Keep tests independent (ordering of tests should not impact testing);
+2. Test one construct or aspect per test;
+3. Every test should make it obvious what is being tested;
+
+## 5. Pull Requests
+
+* All pull requests must target `main`;
+* Updates to __release branches__ are cherry-picked from `main`;
+
+Requirements to be merged:
+
+* The source branch must be __up-to-date__ with `main`;
+* All __commits__ must be __signed__;
+* All GitHub __action checks must pass__;
+* __Documentation and Specs__ must be __updated__ accordingly;
+* PR must be __reviewed and approved__;
+
+### PR Title & Description
+
+Title format: `<impact>: <issue #> - <short description>`
+
+Impact:
+
+- _Breaking_ (major version bump)
+- _Feature_ (minor version bump)
+- _Fix_ (patch version bump)
+
+Other acceptable impact indicators are:
+
+- _Docs_
+- _Tests_
```

Comment:
Let's follow the [conventional commits](https://www.conventionalcommits.org/) standard for PR titles, so when we merge it, the commit message will already have the correct format applied.